Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -951,22 +950,27 @@ public extension Purchases {
         return await productsAsync(productIdentifiers)
     }
 
-    @objc(purchaseProduct:withCompletion:)
-    func purchase(product: StoreProduct, completion: @escaping PurchaseCompletedBlock) {
-        purchasesOrchestrator.purchase(product: product, package: nil, completion: completion)
+    @objc(purchaseProduct:metadata:withCompletion:)
+    func purchase(product: StoreProduct,
+                  metadata: [String: String]? = nil,
```

Comment:
Just curious whether a deliberate decision has been made between defaulting to `nil` and empty?

Reply:
This is good to know, thanks! Adding new functions also keeps it binary compatible from a Swift perspective.